### PR TITLE
Update docs for setting up OCIS backend dev env with Phoenix

### PR DESCRIPTION
**Pull request by PVince81**

### CODE REVIEW

**Comment by PVince81** [comment]
- [ ] integrate OCIS acceptance test section: maybe can get rid of the LDAP docker and use the ocis services instead
- [ ] list of minimum services to run ?

**Comment by PVince81** [comment]
- [ ] document that hot reload doesn't work with OCIS backend, and need to restart webpack watch-all to make it pick up changes (ocis-phoenix doesn't need a restart)

File: README.md [hunk 1]
```diff
@@ -13,7 +13,21 @@ In this document, we will refer to the following:
...
+### Building Phoenix
+
+- Run `yarn install` to build core
+- Run `yarn install-all` to install dependencies of all apps and core
+- Run `yarn dist` to build all apps configured in `config.json`
+
+### Backends
+
+Phoenix can run against either ownCloud 10 as backend or OCIS.
+Depending which one you should, please check the matching section:
```
Comment (kulmann) [inline README.md]: s/should/chose

**Comment by LukasHirt** [comment]
What about moving this into developer docs?

**Comment by PVince81** [comment]
> What about moving this into developer docs?

Would be fine, and then link to it.
This means moving also all the test related stuff and limiting the README to just a quick start.

**Comment by PVince81** [comment]
I've moved all the documentation into the docs folder for hugo to pick up and did some restructuring.

Please see docs/buildling-docs.md to find out how to render the docs locally for reviewing in the web browser :-)

**Comment by PVince81** [comment]
Please note that the OCIS setup instructions are still rough, there are some OCIS services to exclude manually, this is something that needs to be solved in the OCIS repo by providing a better command. Once this is ready we can adjust the docs separately.

File: docs/backend-oc10.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+- Set the "server" key to the URL of the ownCloud server including path. If the URL contains a path, please also add a **trailing slash** there.
+- Set the "clientId" key to the **client identifier** as copied from the "User Authentication" section before.
+- Adjust "url" and "authUrl" using the ownCloud server URL as prefix for both
+- Optionally adjust "apps" for the list of apps to be loaded. These match the app names inside the "apps" folder.
+
+## Running Phoenix
+
+- if running from source, make sure to [build Phoenix]({{< ref "building.md" >}}) first
+- optionally provide a custom domain name: `export SERVER_HOST=0.0.0.0:8300`
```
Comment (LukasHirt) [inline docs/backend-oc10.md:70]: Let's drop this since it's not working?

File: docs/backend-ocis.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+- Clone the [ocis-phoenix repository](https://github.com/owncloud/ocis-phoenix) and follow the setup instructions there.
+
+## Setting up Phoenix
+
+- Copy "config.json.sample-ocis" to "config.json" and change the URLs accordingly if `localhost` is not suitable.
+- Set the environment variables as follows:
+   - `export PHOENIX_CHECKOUT=...` and specify the path to the local git checkout of the [Phoenix repository](https://github.com/owncloud/phoenix)
+   - `export PHOENIX_ASSET_PATH=$PHOENIX_CHECKOUT/dist`
+   - `export PHOENIX_WEB_CONFIG=$PHOENIX_CHECKOUT/phoenix/config.json`
```
Comment (LukasHirt) [inline docs/backend-ocis.md:27]: This leads to a path `phoenix/phoenix/`

File: docs/backend-ocis.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+- Run "ocis-phoenix": `./bin/ocis-phoenix --log-level debug server`
+
+## Running Phoenix
+
+- in the Phoenix checkout folder, run `yarn watch-all-ocis`
+- open "https://localhost:9200" and accept the certificate.
```
Comment (LukasHirt) [inline docs/backend-ocis.md:33]: Change this to link to offer the option of directly navigating there?

File: docs/backend-oc10.md (hunk 2 above)
Comment (PVince81) [reply docs/backend-oc10.md:70]: ref https://github.com/owncloud/phoenix/issues/1989

**Comment by PVince81** [comment]
@LukasHirt comments addressed, thanks for the review!

**Comment by LukasHirt** [comment]
Looks good now. Just thinking about one thing - is this a correct place to have the docs on how to build the Hugo docs? Shouldn't it be in some more generic place?

**Comment by PVince81** [comment]
@LukasHirt agreed, but until I find where exactly this belongs I wanted it to be written down instead of staying in my local notes :smile:

**Comment by LukasHirt** [comment]
> agreed, but until I find where exactly this belongs I wanted it to be written down instead of staying in my local notes 😄

Ok, sounds good 👍

**Pull request merged**